src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -73,7 +79,13 @@ public static class ActivatorUtilities
                 throw new InvalidOperationException(message);
             }
 
-            return bestMatcher.CreateInstance(provider);
+            if (largestLengthWithMultipleAcceptedCtors == bestLength)
+            {
+                string? message = $"multiple constructors for type '{instanceType}' were found with length {bestLength}.";
```

Comment:
I know existing strings are hard-coded, but let's put new strings into the resx file.

Reply:
I'll cover that separately when also updating the package versions for `Autofac`, etc.

Reply:
Why? You are adding new strings now - why add the debt?

Reply:
because I wanted to update versions in a separate PR. now it's addressed in this same PR